shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java

```diff
@@ -190,6 +227,135 @@ public ASTNode visitCreateUser(final CreateUserContext ctx) {
     }
 
     // DDLStatement.g4
+    @Override
+    public ASTNode visitCreateTable(final CreateTableContext ctx) {
+        CreateTableStatement result = new CreateTableStatement();
+        TableSegment table = (TableSegment) visit(ctx.tableName());
+        result.setTable(table);
+        result.getAllSQLSegments().add(table);
+        CreateDefinitionClause_Context createDefinitionClause = ctx.createDefinitionClause_();
+        if (null != createDefinitionClause) {
+            for (CreateDefinition_Context createDefinition : createDefinitionClause.createDefinitions_().createDefinition_()) {
+                ColumnDefinitionContext columnDefinition = createDefinition.columnDefinition();
+                if (null != columnDefinition) {
+                    ColumnDefinitionSegment columnDefinitionSegment = createColumnDefinitionSegment(columnDefinition, result);
+                    result.getColumnDefinitions().add(columnDefinitionSegment);
+                    result.getAllSQLSegments().add(columnDefinitionSegment);
+                }
+                ConstraintDefinition_Context constraintDefinition = createDefinition.constraintDefinition_();
+                ForeignKeyOption_Context foreignKeyOption = null == constraintDefinition ? null : constraintDefinition.foreignKeyOption_();
+                if (null != foreignKeyOption) {
+                    result.getAllSQLSegments().add((TableSegment) visit(foreignKeyOption.referenceDefinition_().tableName()));
+                }
+            }
+        }
+        CreateLikeClause_Context createLikeClause = ctx.createLikeClause_();
+        if (null != createLikeClause) {
+            result.getAllSQLSegments().add((TableSegment) visit(createLikeClause));
+        }
+        return result;
+    }
+
+    @Override
+    public ASTNode visitAlterTable(final AlterTableContext ctx) {
+        AlterTableStatement result = new AlterTableStatement();
+        TableSegment table = (TableSegment) visit(ctx.tableName());
+        result.setTable(table);
+        result.getAllSQLSegments().add(table);
+        if (null != ctx.alterDefinitionClause_()) {
```

Comment:
The above suggestion applied to the following visitors. Could you review those rather long visitors for optimizing?

Reply:
@tristaZero I knew your meaning. I have already previously optimized for the visitor, but no any statement contains `TableSegment`, cause that no match for visit**().